server.go

```diff
@@ -198,15 +198,24 @@ func (sc *serverConn) handleStreams() {
 					continue
 				}
 
-				if len(strms) > int(sc.st.maxStreams) {
+				if _, ok = closedStrms[fr.Stream()]; ok && fr.Type() != FramePriority {
+					sc.writeGoAway(fr.Stream(), StreamClosedError, "frame on closed stream")
+					continue
+				}
+
+				if len(strms) >= int(sc.st.maxStreams) {
 					sc.writeReset(fr.Stream(), RefusedStreamError)
 					continue
 				}
 
 				strm = NewStream(fr.Stream(), sc.clientStreamWindow)
 				strms[fr.Stream()] = strm
 
-				// TODO: sc.lastID = strm.ID()
+				if strm.ID() < sc.lastID {
+					sc.writeGoAway(strm.ID(), ProtocolError, "stream id too low")
```

Comment:
Don't you need to `continue` after the writeGoAway?

Reply:
:+1: good catch